classes/module/Module.php

```diff
@@ -2869,7 +2868,13 @@ public function addOverride($classname)
                 do {
                     $uniq = uniqid();
                 } while (class_exists($classname.'OverrideOriginal_remove', false));
-                eval(preg_replace(array('#^\s*<\?(?:php)?#', '#class\s+'.$classname.'(\s+extends\s+([a-z0-9_]+)(\s+implements\s+([a-z0-9_]+))?)?#i'), array(' ', 'class '.$classname.'Override'.$uniq), implode('', $module_file)));
+                $new_override = preg_replace(array('#^\s*<\?(?:php)?#', '#class\s+'.$classname.'(\s+extends\s+([a-z0-9_]+)(\s+implements\s+([a-z0-9_]+))?)?#i'), array(' ', 'class '.$classname.'Override'.$uniq), implode('', $module_file));
+                try {
+                    $parser->parse($new_override);
+                } catch (PhpParser\Error $e) {
+                    throw new Exception(sprintf(Tools::displayError('The override file "%1$s" of the module %2$s is damaged or invalid.'), 'override'.DIRECTORY_SEPARATOR.$path, $this->displayName));
+                }
+                eval($new_override);
```

Comment:
Same here:

	try {
		if(NULL == eval($new_override)) {
			throw new Exception(sprintf(Tools::displayError('The override file "%1$s" of the module %2$s is damaged or invalid.'), 'override'.DIRECTORY_SEPARATOR.$path, $this->displayName));
		}
	} catch (ParseError $e) { // For PHP 7.0+ only will triggered
		throw new Exception(sprintf(Tools::displayError('The override file "%1$s" of the module %2$s is damaged or invalid.'), 'override'.DIRECTORY_SEPARATOR.$path, $this->displayName));
	}